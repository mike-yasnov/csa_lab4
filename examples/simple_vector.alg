// Простой тест векторных операций

function main() {
    print("Тест векторов...");
    
    // Простые арифметические операции для демонстрации
    var a = 10;
    var b = 20;
    var sum = a + b;
    
    print("Результат: ");
    print_number(sum);
    
    print("Векторы будут добавлены позже");
}

main(); 