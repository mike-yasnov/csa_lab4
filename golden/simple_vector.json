{
  "return_code": 0,
  "stdout": "Загрузка программы: /tmp/tempfile/program.bin\nЗагружено 30 инструкций\nЗагрузка данных: /tmp/tempfile/program_data.bin\nЗагружено 118 байт данных\nЗапуск выполнения (максимум 1000000 тактов)...\n\n=== РЕЗУЛЬТАТЫ ВЫПОЛНЕНИЯ ===\nСостояние: halted\nВыполнено инструкций: 3\nЗатрачено тактов: 8\nФинальный PC: 2\n\nВЫВОД ПРОГРАММЫ:\n[208][162][208][181][209][129][209][130] [208][178][208][181][208][186][209][130][208][190][209][128][208][190][208][178]...\n\nПрограмма завершена успешно.\n",
  "stderr": "",
  "exec_log": "Cycle 000002: PC=0000 00(000000) Stack[0] TOS=empty\nCycle 000007: PC=0001 61(000001) Stack[1] TOS=0\nCycle 000008: PC=0002 01(000000) Stack[0] TOS=empty",
  "debug_listing": "INSTRUCTION MEMORY:\nAddress - Hex Code - Mnemonic\n----------------------------------------\n0000 - 00000000 - PUSH\n0002 - 61010000 - OUT 1\n0004 - 01000000 - POP\n0005 - 000A0000 - PUSH 10\n0007 - 001D0000 - PUSH 29\n0009 - 51000000 - STORE\n000A - 00140000 - PUSH 20\n000C - 00210000 - PUSH 33\n000E - 51000000 - STORE\n000F - 001D0000 - PUSH 29\n0011 - 50000000 - LOAD\n0012 - 00210000 - PUSH 33\n0014 - 50000000 - LOAD\n0015 - 10000000 - ADD\n0016 - 00250000 - PUSH 37\n0018 - 51000000 - STORE\n0019 - 00290000 - PUSH 41\n001B - 61010000 - OUT 1\n001D - 01000000 - POP\n001E - 00250000 - PUSH 37\n0020 - 50000000 - LOAD\n0021 - 61000000 - OUT\n0023 - 01000000 - POP\n0024 - 003E0000 - PUSH 62\n0026 - 61010000 - OUT 1\n0028 - 01000000 - POP\n0029 - 44000000 - RET\n002A - 43000000 - CALL\n002C - 01000000 - POP\n002D - 70000000 - HALT\n\nDATA MEMORY:\nAddress - Hex Dump - ASCII\n----------------------------------------\n0000 - D0 A2 D0 B5 D1 81 D1 82 20 D0 B2 D0 B5 D0 BA D1  - ........ .......\n0010 - 82 D0 BE D1 80 D0 BE D0 B2 2E 2E 2E 00 00 00 00  - ................\n0020 - 00 00 00 00 00 00 00 00 00 D0 A0 D0 B5 D0 B7 D1  - ................\n0030 - 83 D0 BB D1 8C D1 82 D0 B0 D1 82 3A 20 00 D0 92  - ...........: ...\n0040 - D0 B5 D0 BA D1 82 D0 BE D1 80 D1 8B 20 D0 B1 D1  - ............ ...\n0050 - 83 D0 B4 D1 83 D1 82 20 D0 B4 D0 BE D0 B1 D0 B0  - ....... ........\n0060 - D0 B2 D0 BB D0 B5 D0 BD D1 8B 20 D0 BF D0 BE D0  - .......... .....\n0070 - B7 D0 B6 D0 B5 00                                - ......\n"
}